File: react-java/backend/src/main/java/com/clt/hrm/core/administration/program/dtos/PermissionDto.java
package com.clt.hrm.core.administration.program.dtos;

import com.clt.hrm.infra.common.dtos.BaseDto;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class PermissionDto extends BaseDto {
	/**
	 * 
	 */
	private static final long serialVersionUID = 8061565606941902045L;
	private String pgmId;
	private String permId;
	private String permCd;
	private String permNm;
}

File: nuxtjs-nestjs/backend/src/apps/administration/programs/dto/permission.dto.ts
import { IsIn, IsOptional, IsString } from 'class-validator';
import { BaseDto } from '@infra/common/dto';

export class PermissionDto extends BaseDto {
  @IsOptional()
  @IsString()
  permId?: string;

  @IsOptional()
  @IsString()
  permCd?: string;

  @IsOptional()
  @IsString()
  permNm?: string;

  @IsOptional()
  @IsString()
  pgmId?: string;
}
